### PR TITLE
WiP: Initial support for the MSVC compiler

### PR DESCRIPTION
**Pull request by mablanchard**
Supporting Windows MSVC compiler would require:
 - [x] Removing GCC/LLVMisms from CMake scripts
 - [ ] Ensuring C++ codes builds with `cl`
 - [ ] Probably more...

### CODE REVIEW

**Comment by mablanchard** [comment]
This is a very early patch that only allows native Windows+MSVC build to at least pass the configuration step. If the project is interested in MSVC support I'll continue to send patches.

**Comment by andrea-iob** [comment]
> This is a very early patch that only allows native Windows+MSVC build to at least pass the configuration step. If the project is interested in MSVC support I'll continue to send patches.

Adding support for native Windows+MSVC builds would be greatly appreciated. Unfortunately I don't have easy access to MSVC, however @roccoarpa or @alessandro-alaia should be able to test and review the changes. If adding support for MSVC requires changes in some bitpit modules, I will gladly help with the required changes.

File: src/communications/communications.cpp [hunk 1]
```diff
@@ -24,7 +24,9 @@
 
 #if BITPIT_ENABLE_MPI==1
 
+#ifndef _WIN32
 #include <unistd.h>
+#endif
```
Comment (andrea-iob) [inline src/communications/communications.cpp:29]: The header is not needed, you can just removed it (tested on gcc and clang).

Comment (mablanchard) [reply src/communications/communications.cpp]: I think the three last commits are general sanity anyway. Feel free to cherry-pick them to master if you want them to land sooner.

Comment (andrea-iob) [reply src/communications/communications.cpp]: I agree that those commits are useful. I will merged them into master.

**Comment by edoardolombardi** [comment]
The closed pulls https://github.com/optimad/bitpit/pull/264 and https://github.com/optimad/bitpit/pull/282 maybe cover also this requests?
Can we close it after the MSVC compliant modifications already merged, @andrea-iob , @mablanchard?

**Comment by andrea-iob** [comment]
I'm closing the pull request. There were no updates from quite some time. Feel free to reopen this pull request or create a new one if there is any progress.

**Pull request closed without merging**